homeassistant/components/remember_the_milk.py

```diff
@@ -0,0 +1,249 @@
+        api.retrieve_token(frob)
+        token = api.token
+        if api.token is None:
+            _LOGGER.error('Failed to register, please try again.')
+            configurator.notify_errors(
+                request_id,
+                'Failed to register, please try again.')
+            return
+        else:
+            stored_rtm_config.set_token(account_name, token)
+            _LOGGER.debug('retrieved new token from server')
+
+            _create_instance(
+                hass, account_name, api_key, shared_secret, token,
+                stored_rtm_config, component, descriptions)
+
+            configurator.request_done(request_id)
+
+    request_id = configurator.async_request_config(
+        '{} - {}'.format(DOMAIN, account_name),
+        callback=register_account_callback,
+        description='You need to log in to Remember The Milk to' +
+        'connect your account. \n\n' +
+        'Step 1: Click on the link "Remember The Milk login"\n\n' +
+        'Step 2: Click on "login completed"',
+        link_name='Remember The Milk login',
+        link_url=url,
+        submit_caption="login completed",
+    )
+
+
+class RememberTheMilkConfiguration(object):
+    """Internal configuration data for RememberTheMilk class.
+
+    This class stores the authentication token it get from the backend.
+    """
+
+    def __init__(self):
+        """Create new instance of configuration."""
+        if os.path.isfile(CONFIG_FILE_NAME):
+            try:
+                _LOGGER.debug('loading configuration from file: %s',
+                              os.path.abspath(CONFIG_FILE_NAME))
+                with open(CONFIG_FILE_NAME, 'r') as config_file:
+                    self._config = json.load(config_file)
+            except ValueError:
+                _LOGGER.error('failed to load configuration file, creating a '
+                              'new one: %s', os.path.abspath(CONFIG_FILE_NAME))
+                self._config = dict()
+        else:
+            self._config = dict()
+
+    def save_config(self):
+        """Write the configuration to a file."""
+        with open(CONFIG_FILE_NAME, 'w') as config_file:
+            json.dump(self._config, config_file)
+
+    def get_token(self, profile_name):
+        """Get the server token for a profile."""
+        if profile_name in self._config:
+            return self._config[profile_name][CONF_TOKEN]
+        return None
+
+    def set_token(self, profile_name, token):
+        """Store a new server token for a profile."""
+        if profile_name not in self._config:
+            self._config[profile_name] = dict()
+        self._config[profile_name][CONF_TOKEN] = token
+        self.save_config()
+
+    def delete_token(self, profile_name):
+        """Delete a token for a profile.
+
+        Usually called when the token has expired.
+        """
+        self._config.pop(profile_name, None)
+        self.save_config()
+
+
+class RememberTheMilk(Entity):
```

Comment:
There is no reason for there to be an entity as part of this component. Just register a service is all you need.

Reply:
For creating new tasks, I agree, I do not need to create an Entity. 

In the future I would like to be able to:
- display list of tasks
- update existing tasks
- resolve existing tasks

For that I probably need to be able to keep a local state and also display something in the home screen. That's why I created an Entity and not just a service...